### PR TITLE
Added Previous Rank in the table

### PR DESCRIPTION
**Pull request by elymination**
Added `Previous Rank` in the table, alongside its table flag (`False` by default).

content::get_previous_season_id has been created to fetch the previous season id.

### CODE REVIEW

**Comment by elymination** [comment]
https://github.com/zayKenyon/VALORANT-rank-yoinker/issues/152

**Comment by elymination** [comment]
I just profiled `Rank.get_rank` and it is far for being free. my changes adds a call to this function and effectively doubles the cost of initializating the players (from ~0.4s to ~0.8s while solo in menus).

do not merge yet, I will try to add some caching for the `Requests.fetch` calls in `class Rank`.

**Comment by elymination** [comment]
[ebe1b62](https://github.com/zayKenyon/VALORANT-rank-yoinker/pull/153/commits/ebe1b62d24796982e704b5ff317ab8f18647a170) takes care of adding the `Requests.fetch` cache, which is per `puuid` and gets invalidated when going from any state to menus.

while I added this to avoid doubling the cost of initialization because of an additionnal `Rank.get_rank` call, this will also speed up the initialization from menu to pregame if in a party (party members already cached from menus) and from pregame to game (team members already cached from pregame).

File: src/rank.py [hunk 1]
```diff
@@ -5,10 +5,23 @@ def __init__(self, Requests, log, content, ranks_before):
...
+    def get_request(self, puuid):
+        if puuid in self.requestMap:
+            return self.requestMap[puuid]
+
+        response = self.Requests.fetch('pd', f"/mmr/v1/players/{puuid}", "get")
+        self.requestMap[puuid] = response
+        return response
+
+    def invalidate_cached_responses(self):
+        print(f"Invalidating cached reponses")
```
Comment (elymination) [inline src/rank.py:19]: gotta remove this debug `print` statement

**Review by zayKenyon — APPROVED** [review]
Thanks for closing these two long-standing issues!

**Pull request merged**